src/rpcblockchain.cpp

```diff
@@ -696,6 +708,22 @@ UniValue getblockchaininfo(const UniValue& params, bool fHelp)
     softforks.push_back(SoftForkDesc("bip65", 4, tip, consensusParams));
     obj.push_back(Pair("softforks",             softforks));
 
+    UniValue upgrades(UniValue::VARR);
+    vector<CUpgrade> ret = getUpgrades();
+    BOOST_FOREACH(CUpgrade &upgrade, ret) {
+        UniValue rv(UniValue::VOBJ);
+        rv.push_back(Pair("upgradeid", upgrade.upgradeId));
+        rv.push_back(Pair("name", upgrade.name));
+        rv.push_back(Pair("status", upgrade.status));
```

Comment:
`NetworkUpgradeState(tip->nHeight, cosensusParams, idx)` returns the info needed here.